pallets/rewards/src/lib.rs

```diff
@@ -472,4 +471,49 @@ impl<T: Config> Pallet<T> {
 		)?;
 		Ok(())
 	}
+
+	pub fn account_info(
+		account_id: T::AccountId,
+		reward_id: u32,
+	) -> polkadex_primitives::rewards::RewardsInfoByAccount<u128> {
+		if let Some(user_reward_info) = <Distributor<T>>::get(reward_id, account_id) {
+			if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+				let mut rewards_claimable: u128 = 0_u128.saturated_into();
+
+				//if initial rewards are not claimed add it to claimable rewards
+				if !user_reward_info.is_initial_rewards_claimed {
+					rewards_claimable =
+						user_reward_info.initial_rewards_claimable.saturated_into::<u128>();
+				}
+
+				//calculate the number of blocks the user can claim rewards
+				let current_block_no: u128 =
+					<frame_system::Pallet<T>>::block_number().saturated_into();
+				let last_reward_claimed_block_no: u128 =
+					user_reward_info.last_block_rewards_claim.saturated_into();
+				let unclaimed_blocks: u128 =
+					min(current_block_no, reward_info.end_block.saturated_into::<u128>())
+						.saturating_sub(last_reward_claimed_block_no);
+
+				// add the unclaimed block rewards to claimable rewards
+				rewards_claimable = rewards_claimable.saturating_add(
+					user_reward_info
+						.factor
+						.saturated_into::<u128>()
+						.saturating_mul(unclaimed_blocks),
+				);
+
+				let reward_info = polkadex_primitives::rewards::RewardsInfoByAccount {
+					claimed: user_reward_info.claim_amount.saturated_into::<u128>(),
+					unclaimed: user_reward_info
+						.total_reward_amount
+						.saturating_sub(user_reward_info.claim_amount)
+						.saturated_into::<u128>(),
+					claimable: rewards_claimable,
+				};
+				return reward_info
+			}
+		}
+		Default::default()
```

Comment:
It's a bit wrong IMO to receive instance of `RewardsInfoByAccount` with fields values:
```
claimed: "0",
unclaimed: "0",
claimable: "0",
```

even if account not found. Misleading behavior... it should return some kind of error that account not found or rewards info is not found...

Reply:
Returning Error instead ✅